DataDog/ecommerce-workshop
# Ability to configure ports for ads and discounts

https://github.com/DataDog/ecommerce-workshop/issues/107
https://github.com/DataDog/ecommerce-workshop/issues/106

This adds the ability to configure the port mappings for `ads-service`, `ads-service-fixed` , `discounts-service`, and `discounts-service-fixed`. The ports are now set in the `.env` file under `deploy/docker-compose`.

New default port for `ads`: 9292
New default port for `discounts`: 8282

All previous image builds still use 5001 and 5002. Those will continue to work with the new docker-compose env vars as long as the ports are updated in the `.env` var.

`docker-compose-local-baseline.yml` contains the configuration needed to support the dynamic port mapping

The `docker build` commands used by the Github Actions will still work because the `ARG` value inside the Dockerfile is set to default to 9292 / 8282

**How to test**
1. Pull down this branch locally, [rebuild frontend code](https://github.com/DataDog/ecommerce-workshop/blob/main/development.md#recreating-the-code) , change the ports in `deploy/docker-compose/.env`, and start the app via the `baseline` compose file e.g. 

```POSTGRES_USER=postgres POSTGRES_PASSWORD=postgres DD_API_KEY=[API KEY] make local-baseline-start```

2. Visit http://localhost and verify that the homepage loads both discounts and ads
3. Run `docker ps -a` and check that the ports for `discounts` and `ads` are reflecting your desired port
